docs/HowToGuides/GettingStarted.md

```diff
@@ -0,0 +1,593 @@
+
+1. On macOS and Windows, IDEs (Xcode and Visual Studio resp.) serve as an
+   easy way to install development dependencies such as a C++ compiler,
+   a linker, header files, etc. The IDE's build system need not be used to
+   build Swift. On Linux, these dependencies are installed by the
+   distribution's package manager.
+2. CMake is a cross-platform build system for C and C++.
+   It forms the core infrastructure used to _configure_ builds of
+   Swift and its companion projects.
+3. Ninja is a low-level build system that can be used to _build_ the project,
+   as an alternative to Xcode's build system. Ninja is somewhat faster,
+   especially for incremental builds, and supports more build environments.
+4. Sccache is a caching tool:
+   If you ever delete your build directory and rebuild from scratch
+   (i.e. do a "clean build"), Sccache can accelerate the new build
+   significantly. There are few things more satisfying than seeing Sccache
+   cut through build times.
+5. `utils/update-checkout` is a script to help you work with all the individual
+   git repositories together, instead of manually cloning/updating each one.
+6. `utils/build-script` (we will introduce this shortly)
+   is a high-level automation script that handles configuration (via CMake),
+   building (via Ninja or Xcode), caching (via Sccache), running tests and more.
+
+> **Pro Tip**: Most tools support `--help` flags describing the options they
+> support. Additionally, both Clang and the Swift compiler have hidden flags
+> (`clang --help-hidden`/`swiftc --help-hidden`) and frontend flags
+> (`clang -cc1 --help`/`swiftc -frontend --help`) and the Swift compiler
+> even has hidden frontend flags (`swiftc -frontend --help-hidden`). Sneaky!
+
+Phew, that's a lot to digest! Now let's proceed to the actual build itself!
+
+### The actual build
+
+1. Make sure you have Sccache running.
+   ```sh
+   sccache --start-server
+   ```
+   (Optional) Sccache defaults to a cache size of 10GB, which is relatively
+   small compared to build artifacts. You can bump it up, say by setting
+   `export SCCACHE_CACHE_SIZE="50G"` in your dotfile(s). For more details,
+   see the [Sccache README][Sccache].
+2. Decide if you would like to build the toolchain using Ninja or using Xcode.
+   - If you use an editor other than Xcode and/or you want somewhat faster builds,
+     go with Ninja.
+   - If you are comfortable with using Xcode and would prefer to use it,
+     go with Xcode.
+   There is also a third option, which is somewhat more involved:
+   [using both Ninja and Xcode](#using-both-ninja-and-xcode).
+3. Build the toolchain with optimizations, debuginfo, and assertions and run
+   the tests.
+   - Via Ninja:
+     ```sh
+     utils/build-script --skip-build-benchmarks \
+       --skip-ios --skip-watchos --skip-tvos --swift-darwin-supported-archs "x86_64" \
+       --cmake-c-launcher="$(which sccache)" --cmake-cxx-launcher="$(which sccache)" \
+       --release-debuginfo --test
```

Comment:
that's super useful, thank you 👍  esp the sccache and --skip stuff I was not aware of.

Reply:
Cheers! 😄